tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+package steps
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	resty "github.com/go-resty/resty/v2"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const RETRY_TIMEOUT = 15 * time.Minute
+const RETRY_INTERVAL = 30 * time.Second
+
+var namespace string
+var repairId string
+var releaseName string
+
+var (
+	Info    = Color("\033[1;33m%s\033[0m")
+	Outline = Color("\033[1;34m%s\033[0m")
+)
+
+func Color(colorString string) func(...interface{}) string {
+	sprint := func(args ...interface{}) string {
+		return fmt.Sprintf(colorString,
+			fmt.Sprint(args...))
+	}
+	return sprint
+}
+
+func timeTrack(start time.Time, name string) {
+	elapsed := time.Since(start)
+	log.Println(Info(fmt.Sprintf("%s took %s", name, elapsed)))
+}
+
+func RunShellCommand(command *exec.Cmd) error {
+	err := command.Run()
+	return err
+}
+
+func RunShellCommandAndGetOutput(command *exec.Cmd) string {
+	var outb bytes.Buffer
+	command.Stdout = &outb
+	err := command.Run()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return string(outb.String())
+}
+
+// Find returns true if val exists in the slice array, false otherwise
+func Find(slice []string, val string) bool {
+	for _, item := range slice {
+		if item == val {
+			return true
+		}
+	}
+	return false
+}
+
+func g(t *testing.T) *gomega.WithT {
+	return gomega.NewWithT(t)
+}
+
+func deployCluster(t *testing.T, customValues string, helmValues map[string]string) {
+	clusterChartPath, err := filepath.Abs("../../charts/k8ssandra")
+	g(t).Expect(err).To(gomega.BeNil())
+
+	customChartPath, err := filepath.Abs("charts/" + customValues)
+	g(t).Expect(err).To(gomega.BeNil())
+
+	helmOptions := &helm.Options{}
+
+	helmOptions = &helm.Options{
+		// Enable traefik to allow redirections for testing
+		SetValues:      helmValues,
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+		ValuesFiles:    []string{customChartPath},
+	}
+
+	releaseName = "k8ssandra"
+	defer timeTrack(time.Now(), fmt.Sprintf("Installing and starting k8ssandra"))
+	err = helm.InstallE(t, helmOptions, clusterChartPath, releaseName)
+	g(t).Expect(err).To(gomega.BeNil(), "Failed installing k8ssandra with Helm")
+	// Wait for cass-operator pod to be ready
+	g(t).Eventually(func() bool {
+		return podWithLabelIsReady(t, "app.kubernetes.io/name=cass-operator")
+	}, RETRY_TIMEOUT, RETRY_INTERVAL).Should(gomega.BeTrue())
+
+	// Wait for CassandraDatacenter to be ready..
+	WaitForCassDcToBeReady(t)
+}
+
+func WaitForCassDcToBeReady(t *testing.T) {
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	k8s.RunKubectl(t, kubectlOptions, "wait", "--for=condition=Ready", "cassandradatacenter/dc1", fmt.Sprintf("--timeout=%s", 15*time.Minute))
```

Comment:
As I mentioned in the other PR, I would prefer using `wait` or `Eventually` here so that we can log some status. 15 minutes (in the worst case scenario) is a long time to wait without any updates.

Reply:
ok, it's going to be more verbose but can be done.

Reply:
I've now removed all uses of `kubectl wait` throughout the integration test suite.